multiversx_sdk/abi/address_value.py

```diff
@@ -42,10 +42,26 @@ def _check_pub_key_length(self, pubkey: bytes) -> None:
             raise ValueError(f"public key (address) has invalid length: {len(pubkey)}")
 
     def set_payload(self, value: Any):
-        pubkey = bytes(value)
+        if isinstance(value, dict):
+            value = cast(Dict[str, str], value)
+            pubkey = self._extract_pubkey_from_dict(value)
+        else:
+            pubkey = bytes(value)
+
         self._check_pub_key_length(pubkey)
         self.value = pubkey
 
+    def _extract_pubkey_from_dict(self, value: Dict[str, str]) -> bytes:
+        bech32_address = value.get("bech32", None)
+        if bech32_address:
+            return Address.new_from_bech32(bech32_address).get_public_key()
```

Comment:
Here, we reference `core` in `abi` - maybe this is the first occasion? However, there should not be any issues with this.

Reply:
This is not the first time we reference `core` in `abi`. It is also used in `abi.py` and `biguint_value.py`.